mvc/models/model-binding.rst

```diff
@@ -1,8 +1,92 @@
+- ``CancelationToken``: Used to cancel activity in asynchronous controllers.
+
+Once model binding is complete, `validation <https://docs.asp.net/projects/mvc/en/latest/models/validation.html>`_ occurs. Default Model binding works great for the vast majority of development scenarios. It is also extensible so if you have unique needs you can customize the built-in behavior.  
+
+Customize model binding behavior with attributes 
+--------------------------------------------------------
+ASP.NET MVC contains several attributes that you can use to direct its default model binding behavior to a different source. For example, you can specify whether binding is required for a field, or if it should never happen at all by using the ``[BindRequired]`` or ``[BindNever]`` attributes. Alternatively, you can override the default value, and specify the model binder's data source, which is what the ``[From*]`` attributes do, such as the ``[FromForm]`` or ``[FromQuery]``. While the topic of model binding extensibility in depth is outside of the scope of this doc, here are a few customizations that you should be aware of:
+
+- ``[BindRequired]``: This attribute forces a model state error if binding cannot occur.
+- ``[BindNever]``: Tells the model binder to never bind to this parameter.
+- ``[FromHeader]``, ``[FromQuery]``, ``[FromRoute]``, ``[FromForm]``: Use these to specify the exact binding source you want to apply.
+- ``[FromServices]``: This attribute uses `dependency injection <https://docs.asp.net/en/latest/fundamentals/dependency-injection.html>`_ to bind parameters from services.
+- ``[FromBody]``: Use the configured formatters to bind data from the request body. The formatter is selected based on content type of the request.
+- ``[ModelBinder]``: Used to override the default model binder, binding source and name.
+
+Binding attributes are very helpful tools when you need to override the default behavior of model binding.
+
+Binding formatted data from the request body
+--------------------------------------------
+Request data can come in a variety of formats including JSON, XML and many others. When you use the [FrommBody] attribute to indicate that you want to bind a parameter to data in the request body, MVC uses a configured set of formatters to handle the request data based on its content type. By default MVC includes a ``JsonInputFormatter`` class for handling JSON data, but you can add additional formatters for handling XML and other custom formats. 
+
+.. Note:: The ``JsonInputFormatter`` is based off of Json.NET.
+
+ASP.NET selects input formatters based on the Content-Type header and the type of the parameter, unless it has a model binding attribute specifying otherwise. If you'd like to use XML or another format you must configure it in the Startup.cs file. You may have to install a NuGet package for the formatter you want. Your startup code should look something like this:
```

Comment:
make Content-Type a link to https://www.w3.org/Protocols/rfc1341/4_Content-Type.html